### ISSUE TITLE
PDF-Viewer: Zoom-Level: Enlargement ist visible.

### ISSUE THREAD

**Issue by tomtailor1983** [AntonyCorbett/OnlyM#346]
When opening a pdf with a changed zoom level in command panel - not default - you can see the enlargement.

**Steps to reproduce the behavior:**
1. Add PDF to OnlyM
2. Open commandpanel and set zoom level to "page"
3. Show PDF on play button.

**Expected behavior**
The zoom level in the PDF viewer is first adjusted and then displayed. So the enlargement is visible.

**Comment by AntonyCorbett** [comment]
@tomtailor1983 I can't reproduce this. There are 3 options in the command panel; "Horizontal Fit", "Vertical Fit" and "Default". It works ok for me.

**Comment by AlainDema** [comment]
> @tomtailor1983Non posso riprodurlo. Ci sono 3 opzioni nel pannello di comando; "Adattamento orizzontale", "Adattamento verticale" e "Predefinito". Funziona bene per me. Hai perso altre impostazioni?

Buon giorno io personalmente nelle impostazioni non ho trovato queste 3 opzioni nel pannello delle impostazioni, c'è qualcosa di simile quando si proietta il file PDF con il "clone" sul monitor principale. Peccato che in Regia c'è una fastidiosa differenza che crea disagio a chi comanda dato che non rimane uguale a quello che viene proiettato.

**Comment by AntonyCorbett** [comment]
@tomtailor1983 Ah, I understand now, and yes, I can see the effect you describe.

**Comment by tomtailor1983** [comment]
Yes, i am so sorry! I had a misstake in my description. :/ The enlargement **is visible**. So, can you first calculate the desired size and start then the change of visibility?

**Comment by AlainDema** [comment]
Purtroppo cambiare la proporzione dei pixel non serve o almeno io non ci sono riuscito. l'immagine che ho allegato presenta lo 
screen shot della videata sia del "clone" a sx che del proiettato sul monitor del podio a dx. Come potete vedere mentre sul monitor podio l'immagine è perfetta e stabile, quella del "clone" in regia no lo è. Da non programmatore ma osservatore propongo un suggerimento: dato che apparentemente il "clone", anche nella parte scrollata ai lati, coprirebbe quasi tutta la pagina dello schermo magari per eliminare quell'effetto strano di movimento non sarebbe possibile avere il popup "clone" a tutto schermo? forse le due immagini rimangono + equivalenti e non si verifica lo strano spostamento a destra e sinistra nel clone.
Colgo l'occasione per evidenziare un altro piccolo problema sul popup "clone" che riscontriamo con quelli che da noi non sono molto tecnologici ovvero la mancanza del pulsante X di uscita infatti la confondono con quella che si ha nella mini barra comandi per lo zoom e pertanto vanno in panico per quando devono uscire dal "Clone" per interrompere la visualizzazione. Personalmente suggerirei più pratico avere il pulsante dello zoom quadrato e tondo separati e visibilmente disattivati (colore viola di default) o attivati (colore giallo se premuto) mentre tutto a destra un pulsante rosso con la X come nelle classiche finestre di windows o rotondo rosso come apple perché nonostante ci sia la combinazione di tasti Alt+C a sinistra usando il programma non settimanalmente alcuni si dimenticano e rimangono bloccati.
![2023-08-15 differenza tra clone e proiezione](https://github.com/AntonyCorbett/OnlyM/assets/44226662/871c9e79-950e-4aac-968a-fad5595a1c62)
Spero che queste mie considerazioni possano essere utili per arricchire questo UTILISSIMO e FUNZIONALE sistema di proiezione per il secondo monitor. Sono molto felici che sia stato realizzato e mi auguro di utilizzarlo ancora per molto.
Grazie

**Comment by AntonyCorbett** [comment]
@AlainDema Thanks for your feedback. I have made some changes to the "Mirror" window in the latest pre-releases to make the Alt-Z key more prominent. It's not ideal, but a little better.

**Comment by AntonyCorbett** [comment]
@tomtailor1983 I have investigated the issue. We use Chromium's PDF viewer to display PDFs. It renders the PDF and then modifies the layout according to any "view" parameters passed in. Unfortunately, there is no way to detect when it has completed its layout changes. We can add a delay, but the amount of time required for the layout changes depends on the number of pages and other system factors, so adding a delay would be based on an unreliable heuristic. Perhaps the best way forward is to render PDFs using pdf.js but this is a large change for a niche cosmetic issue. So I will close as "not planned" for now.
